library/common/jni/jni_utility.cc

```diff
@@ -58,6 +58,19 @@ envoy_data array_to_native_data(JNIEnv* env, jbyteArray j_data) {
   return {data_length, native_bytes, free, native_bytes};
 }
 
+jbyteArray native_data_to_array(JNIEnv* env, envoy_data data) {
+  jbyteArray j_data = env->NewByteArray(data.length);
```

Comment:
I feel this would be confusing since the user of this method is responsible for deleting the local reference created here, right?

Reply:
That's actually the convention we adopt at the bridge layer for our c types. There's no precedent for JNI LocalRefs but it's not much of a departure.

Reply:
It should at least be mentioned in the documentation for the function.